.circleci/config.yml

```diff
@@ -8,28 +8,113 @@ aliases:
   - &attach_workspace
       attach_workspace:
         at: ~/repo
-
+  - &persist_repo_to_workspace
+      persist_to_workspace:
+        root: ~/repo
+        paths: .
+  - &persist_screenshots_to_workspace
+      persist_to_workspace:
+        root: ~/repo
+        paths:
+          - screenshots
+  - &restore_cache
+      restore_cache:
+        keys:
+          - generic-tools-cache-{{ checksum "package.json" }}
+  - &save_cache
+      save_cache:
+        paths:
+         - ~/.npm
+         - ~/.rustup
+         - ~/.cargo
+        key: generic-tools-cache-{{ checksum "package.json" }}
+  - &setenv
+      run: bash .circleci/set_dynamic_env.sh
+  - &global_environment
+      # runtime versions
+      RUST_VERSION: 1.32.0
+      RUST_TARGETS: wasm32-unknown-unknown
+      # misc
+      CHROME_CODEBASE_URL: https://www.sample.com/dw/yoroi-extension.crx
+      CHROME_WEBSTORE_API_ENDPOINT: https://www.googleapis.com/upload/chromewebstore/v1.1
+      CHROME_DEV_APP_ID: febkioblpijdgicianpklkmolpbagnbo
+      CHROME_STG_APP_ID: bbaanljemfingpglehbbpdlhhahliihe
+      CHROME_PRO_APP_ID: fkkmcnnpaaimcaikbjgklcibgncdgejc
+      SCREENSHOT_DIFF_THRESHOLD: 0
+      SCREENSHOT_DIFF_COLOR: yellow
+      BROWSERS: brave chrome firefox
+      GITHUB_BUILD_BRANCHES: develop staging master
+
+      # Encrypted env vars from ui:
+      # AWS_ACCESS_KEY_ID: "${ARTIFACTS_KEY}"
+      # AWS_SECRET_ACCESS_KEY: "${ARTIFACTS_SECRET}"
+      # AWS_REGION: "${ARTIFACTS_REGION}"
+      # S3_BUCKET: "${ARTIFACTS_BUCKET}"
+      # DOCKER_USERNAME
+      # DOCKER_PASSWORD
+      # GITHUB_PAT
+      # GOOGLE_CLIENT_ID
+      # GOOGLE_CLIENT_SECRET
+      # GOOGLE_REFRESH_TOKEN
+
+#checkout:
+#  post:
 jobs:
   build-setup:
+    environment:
+      <<: *global_environment
+      POC: poc
     working_directory: ~/repo
     docker:
       - image: circleci/node:8-browsers
 
     steps:
       - checkout
+      - *setenv
+      - run:
+          # NOTES: step based on https://discuss.circleci.com/t/show-test-results-for-prospective-merge-of-a-github-pr/1662/8
+          # This is not working on circle's checkout/post hook for some reason, so we moved it here
+          name: Checkout merge-commit if this is a PR build
+          command: |
+            # Get PR number if there is some open PR for develop|staging|master,
+            # even if circle doesn't provide it (triggers are for the branches).
+            # NOTE: we could have multiple PRs from the same branch and this
```

Comment:
multiple OPEN prs, right? If you have multiple closed PRs, everything works fine?

Reply:
Yep!